### ISSUE TITLE
withCredentials deprecated but new docs pointed to don't cover credentials mode

### ISSUE THREAD

**Issue by antgel** [angular/angular#27828]
### Description

I have a [problem](https://stackoverflow.com/questions/53925631/angular-cors-put-missing-authorization-header-after-options-preflight) with my Authorization header not being sent. I've been reading as much documentation as I can find, and [MDN's take](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials) states that:

> For a CORS request with credentials, in order for browsers to expose the response to frontend JavaScript code, both the server (using the Access-Control-Allow-Credentials header) and the client (by setting the credentials mode for the XHR, Fetch, or Ajax request) must indicate that they’re opting in to including credentials.

So, how do I "set the credentials mode" for my XHR PUT request from Angular, if `withCredentials: true` is deprecated?

### What's the affected URL?**

https://angular.io/api/http/RequestOptions (deprecated), pointing to https://angular.io/guide/http which is lovely and shiny but doesn't mention anything about how to set the credentials mode.

**Comment by CDDelta** [comment]
The entirety of `@angular/http` has been deprecated, not just `withCredentials`. You should be moving to the newer package, `@angular/common/http`.

Also, please refrain from posting support questions as issues. Use the Google Group or other channels instead.

**Comment by antgel** [comment]
We are using `@angular/common/http`, hence I don't see the relevance of your reply. This is not a support question in this text, it's pointing out that the documentation or implementation appears to be broken, as something has been deprecated without adequate replacement. Happy to stand corrected.

**Comment by moxival** [comment]
@antgel  https://angular.io/api/common/http/HttpRequest just search the documentation

**Comment by antgel** [comment]
@moxival Thanks for the sane answer. I read the source as you pointed out. It still didn't work for me, the browser popped up a basic authentication dialog even though I was trying to send the `Authorization:` header... So In the end, I did as described [here](https://stackoverflow.com/questions/53925631/angular-cors-put-missing-authorization-header-after-options-preflight#comment94694299_53925631) and used the `fetch()` API, and it worked.

NB The reason I opened this issue was to point out that the documentation is lacking / unclear! :)

**Comment by moxival** [comment]
@antgel yeap, reading the issue again it is clear that you point out a problem with the docs, we are using JWT for our projects without any issue but hopefully someone from the docs team will pick it up.

P.S.
Don't mind the "thumbs down" there are always "know-it all but never-share-it guys" 😄

**Comment by moxival** [comment]
@trotyl  i believe that you should invest less time doing the above and more time reading the issue reports. Yes it might not be up to the "standards" but let's agree that we are not in the army and we do not work for Angular, so instead of discouraging someone to post issues for Angular can we all focus on fixing what is wrong with it and encourage the posting and describing issues correctly?

If you take a look at the HTTP guide (the one the poster links to) https://angular.io/guide/http you will see that there is no mentioning how to use the withCredentials property which is the issue of the original poster. If I can understand that (without the golden and oh! so important issue template) then you should be able to understand it as well. So just to make it clear, the poster does not know how to use the withCredentials, because it is not in the documentation and if it is in the docs, it is not easily found - and that my friend is the issue in its essence!

And the last thing and probably the most important one, stop being so arrogante in this community and get down of your high horse. Maybe browse other open source projects and their issue board and you might earn how to respect your fellow developers.

Merry Christmas @trotyl

**Comment by trotyl** [comment]
@moxival 

> can we all focus on fixing what is wrong with it and encourage the posting and describing issues correctly

Yep, I mentioned clearly what exact what's wrong in this issue and what should be done:

> you didn't fill the template of document request

> all of your contents are misleading, you're asking why a long-deprecated package is deprecated

> if you think current document doesn't clearly reflect that, please describe exactly what should be improved

---

> If you take a look at the HTTP guide (the one the poster links to) https://angular.io/guide/http you will see that there is no mentioning how to use the withCredentials property which is the issue of the original poster.

I don't mind a document request to credentials integration, but as explained above, this isn't a document request.

Also, for any kind issue a search for existing one is required, but the OP fails it again: https://github.com/angular/angular/issues/27579

---

> And the last thing and probably the most important one, stop being so arrogante in this community and get down of your high horse.

Could you explain what is `this community`? The issue channel? I'm deliberately against any bad-formed support request here, but I'll try my best to help with good issues (mostly to be bug reports, as documentation request doesn't quite need my help).

---

> Maybe browse other open source projects and their issue board and you might earn how to respect your fellow developers.

Most project has provided IRC channel for support request including Angular, that's why our Gitter channel exists: https://gitter.im/angular/angular

**Comment by antgel** [comment]
@trotyl You didn't invent the Internet, nor netiquette, nor open-source, nor are you immune from criticism in the same way that you dish it out. You have missed the mark so widely here on so many levels, that I don't know where to start - probably a good thing. If you have anything further to add to this issue, please make it constructive, otherwise I would respectfully request that you move on and leave it for those who want to make the docs better (documenting use of `withCredentials` and friends in an `HttpClient` context.

**Comment by moxival** [comment]
Good job @trotyl! Good job! 

See, not that hard! 😸 Geez some people.....

**Comment by trotyl** [comment]
Sorry for anything sounds to be aggressive. I can provide some details (in different wording) here.

> I have a problem with my Authorization header not being sent.

Given what you posted in the StackOverflow:

```typescript
const headers = new HttpHeaders();                                                                                                                                                  
headers.set('Authorization', 'basic ' + btoa(environment.eventUsername + ':' + environment.eventPassword));                                                                         
headers.set('Content-Type', 'application/json');
```

This is wrong as `HttpHeaders` immutable, you'll end up sending an empty `HttpHeaders` object, this isn't related to `Authorization`.

> So, how do I "set the credentials mode" for my XHR PUT request from Angular, if withCredentials: true is deprecated?

`withCredentials` is not deprecated, but `@angular/http` package is, I'd suggest change it for others to understand what's the point for this issue.

> https://angular.io/api/http/RequestOptions (deprecated), pointing to https://angular.io/guide/http which is lovely and shiny but doesn't mention anything about how to set the credentials mode.

`RequestOptions` also wasn't being linked in current doc, if any article or forum did, I'd suggest leave a reply for updating the solution.

The request for adding `withCredentials` in HTTP guide is valid, feel free to reopen, but please make the request clear like:

> Adding an example with `withCredentials` usage in the Http Guide.

---

Merry Christmas
